src/components/AppModal.vue

```diff
@@ -1,38 +1,137 @@
-      icon="times"
-      @click="close"
-      v-tooltip="'Close dialog (esc)'"
-    />
-  </VueFinalModal>
+  <teleport to="body">
+    <transition name="fade">
+      <div
+        v-if="modelValue"
+        class="overlay"
+        @mousedown="close"
+        @touchstart="close"
+        role="dialog"
+        aria-modal="true"
+        :aria-label="label"
+      >
+        <div ref="modal" class="modal" @mousedown.stop @touchstart.stop>
+          <AppButton
+            class="close"
+            icon="times"
+            @click="close"
+            v-tooltip="'Close dialog (esc)'"
+          />
+          <slot />
+        </div>
+      </div>
+    </transition>
+  </teleport>
 </template>
 
 <script lang="ts">
-import { defineComponent } from "vue";
+import { defineComponent, nextTick } from "vue";
+import { disableBodyScroll, enableBodyScroll } from "body-scroll-lock";
 
-// modal component with arbitrary content
+// basic modal component with arbitrary content
 export default defineComponent({
   props: {
     // open state
     modelValue: Boolean,
+    // modal aria label
+    label: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      originalFocus: null as HTMLElement | null,
+    };
   },
   emits: ["update:modelValue"],
+  methods: {
+    // update model value to close modal
+    close() {
+      this.$emit("update:modelValue", false);
+    },
+    // on key down
+    keyDown(event: KeyboardEvent) {
+      if (event.key === "Escape") this.close();
+    },
+  },
+  mounted() {
+    // global keydown listener
+    window.addEventListener("keydown", this.keyDown);
+  },
+  updated() {
+    // modal element
+    const modal = this.$refs.modal as HTMLElement;
+    // rest of app besides modal
+    const app = document.querySelector("#app");
+
+    // if modal just opened
+    if (this.modelValue) {
+      // hide for screen readers
+      app?.setAttribute("aria-hidden", "true");
+      // disable focus
+      app?.setAttribute("inert", "true");
+      // disable body scroll
+      disableBodyScroll(modal);
+      // focus first focusable element in modal
+      const query = "input, textarea, button, select";
+      const focusable = modal.querySelector(query) as HTMLElement;
+      focusable?.focus();
+    }
+  },
+  beforeUpdate() {
+    // modal element
+    const modal = this.$refs.modal as HTMLElement;
+    // rest of app besides modal
+    const app = document.querySelector("#app");
+
+    // if modal about to be opened
+    if (this.modelValue) {
+      this.originalFocus = document.activeElement as HTMLElement;
+    }
+    // if modal about to be closed
+    else {
+      // show for screen readers
+      app?.removeAttribute("aria-hidden");
+      // enable focus
+      app?.removeAttribute("inert");
+      // enable body scroll
+      enableBodyScroll(modal);
+      // restore focus to what had focus before modal opened
+      nextTick(() => this.originalFocus?.focus());
```

Comment:
Just to make sure, is the desired behavior for it to refocus the feedback button after dismissing the modal? That'd make sense, but I could also see wanting to refocus the thing they had focused before they clicked the feedback button.